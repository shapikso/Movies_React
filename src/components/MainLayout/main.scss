.header__container {
  padding: 30px 0;
  display: flex;
  justify-content: space-between;
  align-items: center;
}

// @font-face {
//   font-family: 'Roboto';
//   src: url("../../fonts/Roboto-Light.ttf") format('truetype');
//   font-weight: 300;
// }

// @font-face {
//   font-family: 'Roboto';
//   src: url("../../fonts/Roboto-Regular.ttf") format('truetype');
//   font-weight: 400;
// }

// @font-face {
//   font-family: 'Roboto';
//   src: url("../../fonts/Roboto-Bold.ttf") format('truetype');
//   font-weight: 700;
// }
