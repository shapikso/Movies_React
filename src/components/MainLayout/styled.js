import styled from "styled-components";

export const StContainer = styled.div`
  max-width: 1500px;
  padding: 0 40px;
  margin: 0 auto;
`;

export const StHeaderContainer = styled.div`
  padding: 30px 0;
  display: flex;
  justify-content: space-between;
  align-items: center;
`;

export const StHeaderLink = styled.div`
  display: flex;
  justify-content: space-between;
`;

export const StButtonLinkWrapper = styled.div`
  width: 100px;
  height: 40px;
  color: var(--light-primary-color);
  letter-spacing: 0.1em;

  display: flex;
  justify-content: center;
  align-items: center;

  border: 1px solid var(--light-primary-color);
  border-radius: 5px;

  padding: 5px;

  transition: all 0.25s ease;

  :active {
    background-color: var(--light-primary-color-opacity);
    border-color: transparent;
  }

  :hover {
    background-color: var(--light-primary-color);
    color: var(--dark-primary-color);
  }
  button{
    padding: 5px;
  }
  button: hover {
    background-color: var(--light-primary-color);
    color: var(--dark-primary-color);
  }
`;