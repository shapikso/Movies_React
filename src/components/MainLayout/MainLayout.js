import React, {Component} from 'react';
import './main.scss';
import Logo from './Logo/Logo';
import {Link} from "react-router-dom";
import Button from '../common/Button/Button';

class MainLayout extends Component {
    constructor(props) {
        super(props);
    }

    render() {
        return (
            <div>
                <header className="header">
                    <div className="container">
                        <div className="header__container">
                            <div className="header__logo">
                                <Link to="/movies">
                                    <Logo/>
                                </Link>
                            </div>
                            <div className="header__links">
                                <div className="basic-link header__links-item">
                                    <Link to="/sing-up">
                                        <Button className="button--link">Sing up</Button>
                                    </Link>
                                </div>
                            </div>
                        </div>
                    </div>
                </header>
                <main className="main">{this.props.children}</main>
            </div>
        );
    }
}

export default MainLayout;