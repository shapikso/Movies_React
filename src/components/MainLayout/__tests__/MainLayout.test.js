import React from 'react';
import {mount, shallow} from 'enzyme';
import MainLayout from "../MainLayout";

jest.mock('react-router-dom', () => ({
    useLocation: () => ({ pathname: '/' }),
    Link: "a",
}));

describe('MainLayout', () => {

    const props={
        children: <div id="main-component"/>,
    };

    it('should render correctly', () => {
        const component = shallow(<MainLayout {...props}/>);
        expect(component).toMatchSnapshot();
    });
    it('should render children', () => {
        const component = mount(<MainLayout {...props}/>);
        expect(component.find('#main-component')).toHaveLength(1);
    });
    it('should render 2 links', () => {
        const component = mount(<MainLayout {...props}/>);
        expect(component.find('a')).toHaveLength(2);
        expect(component.find('a').at(1).getElement().props.to).toEqual('/sing-up');
        expect(component.find('button').text()).toEqual('Sing Up');
    });

});