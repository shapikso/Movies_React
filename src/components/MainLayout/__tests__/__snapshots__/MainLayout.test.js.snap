// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainLayout should render correctly 1`] = `ShallowWrapper {}`;
